express/src/middleware/urlValidator.ts

```diff
@@ -0,0 +1,10 @@
+import {NextFunction, Request, Response} from "express";
+
+type MiddlewareFunction<T, U, V> = (T: Request, U: Response, V: NextFunction) => void;
+const URL = require("url").URL;
+export function isUriValid(uri: string) {
+    if (!uri.startsWith("http://") && !uri.startsWith("https://")) {
```

Comment:
As also discussed, the URL class needs to be used and the protocol property compared, only allowing http for the special case that the hostname is localhost.